src/util/utils.ts

```diff
@@ -79,3 +80,7 @@ export const validateEmail = (email: string) => {
       /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
     )
 }
+const base58 = basex(
```

Comment:
In the core, we export this thingy:

```js
export const base58 = {
  parse(text: string): Uint8Array {
    return base58Codec.decode(text)
  },
  stringify(data: Uint8Array | number[]): string {
    return base58Codec.encode(data)
  }
}
```

Could we make the login-ui do the same thing, so we can more easily copy & paste between repos?

Reply:
Hope this isn't overkill:

```ts
const base58Codec: BaseConverter = basex(
  '123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz'
)
interface Base58 {
  parseUnsafe: (text: string) => Uint8Array | undefined
  parse: (text: string) => Uint8Array
  stringify: (data: Uint8Array | number[]) => string
}

export const base58: Base58 = {
  parseUnsafe(text: string): Uint8Array | undefined {
    return base58Codec.decodeUnsafe(text)
  },
  parse(text: string): Uint8Array {
    return base58Codec.decode(text)
  },
  stringify(data: Uint8Array | number[]): string {
    return base58Codec.encode(data)
  }
}
```

Reply:
Yes, it's overkill. The type definition isn't needed - TypeScript infers this, since it's a singleton. The `parseUnsafe` is a fine addition. The core doesn't have it (we use try / catch around the normal parse), but it's not going to cause an problems.